### PR TITLE
Display categories on sideBar

### PR DESCRIPTION
**Pull request by benrcd**
A global methode was created to provide the sideBar with categories all the time.
A collapse menu is used to display said categories.

### CODE REVIEW

**Review by Spolito24 — DISMISSED** [review]

File: templates/component/_sideNavbar.html.twig [hunk 1]
```diff
@@ -1,13 +1,29 @@
...
+                <div class="overflow-y-auto">
+                    <button class="btn btn-primary" type="button" data-bs-toggle="collapse" data-bs-target="#perimeterList" aria-expanded="false" 
+                    aria-controls="perimeterList">
+                        Catégories
+                    </button>
+                    <div class="collapse" id="perimeterList">
+                        {% for perimeter in perimeters.getPerimeters %}
+                        <div class="card"><a class="dropdown-item" href="#">{{perimeter.perimeter}}</a>
+                        </div>
+                        {% endfor %}
```
Comment (aldaitzwild) [inline templates/component/_sideNavbar.html.twig]: I don't understand what you intend to show here. Do you realize that you load in global and show here ALL the ideas of your database. If you have 5000 ideas, there will be 5000 cards here ...

**Review by aldaitzwild — CHANGES_REQUESTED** [review]

**Comment by benrcd** [comment]
The function $getPerimeters was replaced by the function $getDistinctPerimeters so the displayed list won't be filled with clone, only the ones available in the idea form.

File: templates/component/_sideNavbar.html.twig [hunk 2]
```diff
@@ -1,13 +1,29 @@
...
             </a>
+
+                <div class="overflow-y-auto">
+                    <button class="btn btn-primary" type="button" data-bs-toggle="collapse" data-bs-target="#perimeterList" aria-expanded="false" 
+                    aria-controls="perimeterList">
+                        Catégories
+                    </button>
+                    <div class="collapse" id="perimeterList">
+                        {% for perimeter in perimeters.getDistinctPerimeters %}
+                        <div class="card"><a class="dropdown-item" href="#">{{perimeter.perimeter}}</a>
```
Comment (aldaitzwild) [inline templates/component/_sideNavbar.html.twig]: Better ! But still too much. You don't need to get here full Idea objects. getDistinctPerimeters should return an array of string, not an array of ideas

**Review by aldaitzwild — CHANGES_REQUESTED** [review]

**Comment by benrcd** [comment]
The function getDistinctPerimeters focus now on the strings corresponding to the differents 'perimeter' instead of using the whole object all the time. It should be lighter and straight to the point

**Review by AurelOconnell — APPROVED** [review]
it seems fine to me, shorter and lighter, array_map returns only $result from array perimeter. But as I said before, difficult for me to be really sure about the whole thing

**Review by aldaitzwild — APPROVED** [review]

**Pull request merged**